routing/road_graph_router.cpp

```diff
@@ -212,16 +212,17 @@ IRouter::ResultCode RoadGraphRouter::CalculateRoute(m2::PointD const & startPoin
   m_roadGraph->AddFakeEdges(startPos, startVicinity);
   m_roadGraph->AddFakeEdges(finalPos, finalVicinity);
 
-  vector<Junction> path;
+  RoutingResult<Junction> result;
   IRoutingAlgorithm::Result const resultCode =
-      m_algorithm->CalculateRoute(*m_roadGraph, startPos, finalPos, delegate, path);
+      m_algorithm->CalculateRoute(*m_roadGraph, startPos, finalPos, delegate, result);
 
   if (resultCode == IRoutingAlgorithm::Result::OK)
   {
-    ASSERT(!path.empty(), ());
-    ASSERT_EQUAL(path.front(), startPos, ());
-    ASSERT_EQUAL(path.back(), finalPos, ());
-    ReconstructRoute(move(path), route, delegate);
+    ASSERT(!result.path.empty(), ());
+    ASSERT_EQUAL(result.path.front(), startPos, ());
+    ASSERT_EQUAL(result.path.back(), finalPos, ());
+    ASSERT_GREATER(result.distance, 0., ());
```

Comment:
The path must contains at least the start and the end points of the route if routing possible and result is OK. Other algorithms supposes that.